File: web/src/api/sysMenu.ts
import { http,resData } from "@/utils/http";
 

// @Tags SysMenu
// @Summary 创建SysMenu
export const createSysMenu = (data?: object) => {
  return http.request<resData>("post", "/sysMenu/createSysMenu", { data });
};

// @Tags SysMenu
// @Summary 删除SysMenu
export const deleteSysMenu = (data?: object) => {
  return http.request<resData>("post", "/sysMenu/deleteSysMenu", { data });
};


// @Tags SysMenu
// @Summary 删除SysMenu
export const deleteSysMenuByIds = (data?: object) => {
  return http.request<resData>("post", "/sysMenu/deleteSysMenuByIds", { data });
};


// @Tags SysMenu
// @Summary 更新SysMenu
export const updateSysMenu = (data?: object) => {
  return http.request<resData>("post", "/sysMenu/updateSysMenu", { data });
};


// @Tags SysMenu
// @Summary 用id查询SysMenu
export const findSysMenu =  (params?: object) => {
  return http.request<resData>("get", "/sysMenu/findSysMenu", { params });
};


// @Tags SysMenu
// @Summary 分页获取SysMenu列表
export const getSysMenuList =  (params?: object) => {
  return http.request<resData>("get", "/sysMenu/getSysMenuList", { params });
};




// @Tags SysMenu
// @Summary 快速编辑
export const quickEdit = (data?: object) => {
  return http.request<resData>("post", "/sysMenu/quickEdit", { data });
};


// @Tags SysMenu
// @Summary 导出excelSysMenu列表
export const excelList =(data?: object) => {
  return http.request<resData>("get", "/sysMenu/excelList", { data });
};

File: server/router/business/sysMenu.go
package business

import (
	"go-cms/api/v1"
	"go-cms/middleware"
	"github.com/gin-gonic/gin"
)

type SysMenuRouter struct {
}

// InitSysMenuRouter 初始化 SysMenu 路由信息
func (s *SysMenuRouter) InitSysMenuRouter(Router *gin.RouterGroup) {
	router := Router.Group("sysMenu").Use(middleware.OperationRecord())
	routerNoRecord := Router.Group("sysMenu")
	var apiV1 = v1.ApiGroupApp.BusinessApiGroup.SysMenuApi
	{
		router.POST("createSysMenu", apiV1.CreateSysMenu)   // 新建SysMenu
		router.POST("deleteSysMenu", apiV1.DeleteSysMenu) // 删除SysMenu
		router.POST("deleteSysMenuByIds", apiV1.DeleteSysMenuByIds) // 批量删除SysMenu
		router.POST("updateSysMenu", apiV1.UpdateSysMenu)    // 更新SysMenu
	    router.POST("quickEdit", apiV1.QuickEdit)  // 快速编辑	
	}
	{
		routerNoRecord.GET("findSysMenu", apiV1.FindSysMenu)        // 根据ID获取SysMenu
		routerNoRecord.GET("getSysMenuList", apiV1.GetSysMenuList)  // 获取SysMenu列表
        routerNoRecord.GET("excelList", apiV1.ExcelList)  // 分页导出excel SysMenu列表
	}
}
